### PR TITLE
Fix multiple issues with ivy.jac

### PR DESCRIPTION
**Pull request by somaia02**
Update ivy.jac to:
1. Convert x_in to ivy_array since the function passed to ivy.jac expects the input to be ivy arrays and not native array
2. Accept functions that return named tuples
3. use jacrev instead of jacfwd since the results are different in some cases
4. For paddle: 
   * Return ivy.array instead of a paddle.jacobian object
   * Return the jacobian with the right shape

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/backends/jax/gradients.py [hunk 1]
```diff
@@ -145,10 +145,15 @@ def stop_gradient(
...
 def jac(func: Callable):
-    grad_fn = lambda x_in: ivy.to_native(func(x_in), nested=True)
+    grad_fn = lambda x_in: ivy.to_native(
+        func(ivy.to_ivy(x_in, nested=True)),
+        nested=True,
+        include_derived={tuple: True},
+    )
     callback_fn = lambda x_in: ivy.to_ivy(
-        jax.jacfwd(grad_fn)((ivy.to_native(x_in, nested=True))),
+        jax.jacrev(grad_fn)((ivy.to_native(x_in, nested=True))),
```
Comment (vedpatwardhan) [inline ivy/functional/backends/jax/gradients.py:154]: I think there's some cases where `jacrev` is consistent with the other backends and others where `jacfwd` is more consistent. I had also created an [issue](https://github.com/google/jax/issues/14149) for this, but doesn't seem to have been responded to.

File: ivy/functional/backends/jax/gradients.py [hunk 2]
```diff
@@ -145,10 +145,15 @@ def stop_gradient(
...
         nested=True,
+        include_derived={tuple: True},
```
Comment (vedpatwardhan) [inline ivy/functional/backends/jax/gradients.py:156]: Good catch! `include_derived=True` would be even more generalized in that case, so that we cover other derived types like `ivy.Container`. Same applies to the other backends

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @somaia02, just requested a couple of changes, I'll also request @Ishticode's review. Thanks for looking into this 😄

File: ivy/functional/backends/jax/gradients.py (hunk 1 above)
Comment (somaia02) [reply ivy/functional/backends/jax/gradients.py]: It's a problem with torch.jacfwd as well. I only tried with the `eigh` function though. It doesn't return nan but returns different values in each case.

**Comment by somaia02** [comment]
Regarding the issue with computing higher-order gradients. I tried this code and it works well. Are there any other cases that fail?
```python
x = ivy.array(1.)
fn = lambda x: ivy.jac(ivy.sin)(x)
j = ivy.jac(fn)(x)
print(j) # ivy.array(-0.84147096)
```

File: ivy/functional/backends/jax/gradients.py (hunk 1 above)
Comment (vedpatwardhan) [reply ivy/functional/backends/jax/gradients.py]: oh okay got it! Thanks for clarifying, we can replace `jacrev` by `jacfwd` if it's more consistent for the cases we're testing 👍

**Comment by vedpatwardhan** [comment]
> Regarding the issue with computing higher-order gradients. I tried this code and it works well. Are there any other cases that fail?
> 
> ```python
> x = ivy.array(1.)
> fn = lambda x: ivy.jac(ivy.sin)(x)
> j = ivy.jac(fn)(x)
> print(j) # ivy.array(-0.84147096)
> ```

I can't really think of a specific case, but I'll leave it with @Ishticode to test it out 😄

**Comment by somaia02** [comment]
Hello @Ishticode Can you please review this? Thank you.

**Review by Ishticode — APPROVED** [review]
Hi @somaia02 
Thanks for the great work here.
the chaining above works with ivy in accordance with jax in following code
```python
import jax
import jax.numpy as jnp
x = jnp.array(1.)
fn = lambda x: jax.jacrev(jnp.sin)(x)
j = (jax.jacrev(fn))(x)
print(j)
``` 

and any other similar combinations.

Although I am not sure we are testing for higher orders unless i missed something. Do you want to create another PR to add the additional tests @somaia02 and we merge this (which is easier) or continue on.

**Comment by somaia02** [comment]
Okay. I will merge this and make another PR with the additional tests. Thank you.
